bin/wait_for_it.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# little script that simply watches for the secret puller to be done.  
+# script will return 0 on success, 1 on timeout (we'll give the puller 60s), and 2 for any other error
+
+if [ -z ${DONE_FILE} ]; then
+	DONE_FILE=/secrets/.done 
+fi
+
+if [ -z ${TIMEOUT} ]; then
+	TIMEOUT=60
+fi
+
+function exit_success {
+  exit 0
+}
+
+function exit_timeout {
+  echo "Timeout waiting for secrets"
+  exit 1
+}
+
+function exit_error {
+  exit 2
+}
+
+while [ $TIMEOUT -gt 0 ]; do
+  TIMEOUT=$(($TIMEOUT - 1))
+  sleep 1
+  if [ -f $DONE_FILE ]; then
```

Comment:
start with the check and then sleep so the script can be done instantly in the ideal case